tests/net/context/src/main.c

```diff
@@ -66,87 +67,60 @@ static struct k_sem wait_data;
 #define MY_PORT 1969
 #define PEER_PORT 16233
 
-static bool net_ctx_get_fail(void)
+static void net_ctx_get_fail(void)
 {
 	struct net_context *context;
 	int ret;
 
 	ret = net_context_get(AF_UNSPEC, SOCK_DGRAM, IPPROTO_UDP, &context);
-	if (ret != -EAFNOSUPPORT) {
-		TC_ERROR("Invalid family test failed (%d)\n", ret);
-		return false;
-	}
+	zassert_equal(ret, -EAFNOSUPPORT,
+			"Invalid family test failed");
 
 	ret = net_context_get(AF_INET6, 10, IPPROTO_UDP, &context);
-	if (ret != -EPROTOTYPE) {
-		TC_ERROR("Invalid context type test failed (%d)\n", ret);
-		return false;
-	}
+	zassert_equal(ret, -EPROTOTYPE,
+			"Invalid context type test failed ");
 
 	ret = net_context_get(AF_INET6, SOCK_DGRAM, IPPROTO_ICMPV6, &context);
-	if (ret != -EPROTONOSUPPORT) {
-		TC_ERROR("Invalid context protocol test failed (%d)\n", ret);
-		return false;
-	}
+	zassert_equal(ret, -EPROTONOSUPPORT,
+			"Invalid context protocol test failed");
 
 	ret = net_context_get(1, SOCK_DGRAM, IPPROTO_UDP, &context);
-	if (ret != -EAFNOSUPPORT) {
-		TC_ERROR("Invalid context family test failed (%d)\n", ret);
-		return false;
-	}
+	zassert_equal(ret, -EAFNOSUPPORT,
+			"Invalid context family test failed");
 
 	ret = net_context_get(AF_INET6, SOCK_STREAM, IPPROTO_TCP, &context);
-	if (ret != -EPROTOTYPE) {
-		TC_ERROR("Invalid context proto type test failed (%d)\n", ret);
-		return false;
-	}
+	zassert_equal(ret, -EPROTOTYPE,
+			"Invalid context proto type test failed");
 
 	ret = net_context_get(AF_INET6, SOCK_DGRAM, IPPROTO_TCP, &context);
-	if (ret != -EPROTONOSUPPORT) {
-		TC_ERROR("Invalid context proto value test failed (%d)\n", ret);
-		return false;
-	}
+	zassert_equal(ret, -EPROTONOSUPPORT,
+			"Invalid context proto value test failed");
 
 	ret = net_context_get(AF_INET6, SOCK_DGRAM, IPPROTO_UDP, NULL);
-	if (ret != -EINVAL) {
-		TC_ERROR("Invalid context value test failed (%d)\n", ret);
-		return false;
-	}
-
-	return true;
+	zassert_equal(ret, -EINVAL,
+			"Invalid context value test failed ");
 }
 
-static bool net_ctx_get_success(void)
+static void net_ctx_get_success(void)
 {
 	struct net_context *context = NULL;
 	int ret;
 
 	ret = net_context_get(AF_INET6, SOCK_DGRAM, IPPROTO_UDP, &context);
-	if (ret != 0) {
-		TC_ERROR("Context get test failed.\n");
-		return false;
-	}
-
-	if (!context) {
-		TC_ERROR("Got NULL context\n");
-		return false;
-	}
+	zassert_equal(ret, 0,
+			"Context get test failed");
+	zassert_not_null(context, "Got NULL context");
 
 	ret = net_context_put(context);
-	if (ret != 0) {
-		TC_ERROR("Context put test failed.\n");
-		return false;
-	}
+	zassert_equal(ret, 0,
+			"Context put test failed");
 
 	if (net_context_is_used(context)) {
-		TC_ERROR("Context put check test failed.\n");
-		return false;
+		zassert_true(0, "Context put check test failed");
```

Comment:
Wouldn't it make more sense to do zassert_false(net_context_is_used(context));

Reply:
agree